code/modules/vtr13/mobs/human/handle_ghouling.dm

```diff
@@ -0,0 +1,36 @@
+/mob/living/carbon/human/proc/handle_ghouling(mob/living/carbon/human/domitor)
+	if(!src || !domitor)
+		return FALSE
+	
+	if(!ishumanbasic(src) || !iskindred(domitor))
+		return FALSE
+	
+	if(istype(src, /mob/living/carbon/human/npc))
+		var/mob/living/carbon/human/npc/NPC = src
+		NPC.ghoulificate(src)
+		return TRUE
+
+	src.set_species(/datum/species/ghoul)
+	src.regent_clan = domitor.clane
+	src.roundstart_vampire = FALSE
+	var/datum/species/ghoul/g_species = src.dna.species
+	g_species.master = domitor
+	g_species.last_vitae = world.time
+
+	var/response_ghoul = tgui_alert(src, "You have been ghouled! Do you wish to keep being a ghoul on your save slot?", "Confirmation", list("Yes", "No")) == "Yes"
+	if(!response_ghoul)
+		return FALSE
+
+	var/datum/preferences/BLOODBONDED_prefs_g = src.client.prefs
+	if(BLOODBONDED_prefs_g.discipline_types.len == 3)
```

Comment:
not actually sure if we want characters who are ghouled to get disciplines that night; it's not lore-compliant though it's probably more fun to let people play with their toys right away

Reply:
Oh - yeah I can change that. This is leftover and transplanted code from the give vitae process.